src/AwsMetadata.js

```diff
@@ -53,6 +53,9 @@ export default class AwsMetadata {
       this.lookupMetadataKey(`network/interfaces/macs/${results.mac}/vpc-id`, (err, vpcId) => {
         results['vpc-id'] = vpcId;
         this.logger.debug('Found Instance AWS Metadata', results);
+        Object.keys(results)
+              .filter(k => results[k] === null || results[k] === undefined)
+              .forEach(k => delete results[k]);
```

Comment:
Thanks @tbfangel for the updates, and especially the tests you added for this. Instead of calling `delete` here and having to check for null and undefined, it might be a little more straightforward to do:

```javascript
const filteredResults = Object.keys(results).reduce((filtered, current) => {
  if (results[current]) { filtered[current] = results[current]; }
  return filtered;
}, {});
resultsCallback(filteredResults);
```
Apologies for the back-and-forth! This PR looks good otherwise.